src/jdk.javadoc/share/classes/jdk/javadoc/internal/doclets/formats/html/TagletWriterImpl.java

```diff
@@ -612,6 +607,53 @@ private Content linkSeeReferenceOutput(Element holder,
         }
     }
 
+    /*
+     * Given a method, returns the actual method to hyperlink to; might return
+     * the same method if that method can be hyperlinked to directly or an
+     * empty optional if that method is not documented anywhere and thus cannot
+     * be hyperlinked to.
+     *
+     * Why is this needed? Depending on configuration, a declared method does
+     * not always result in a corresponding "Method Details" entry, which we
+     * could link to. To not produce broken links (i.e. links that lead
+     * nowhere), we need to find the method that is a substitute for the
+     * given method.
+     */
+    public Optional<ExecutableElement> getActualMethod(ExecutableElement method) {
```

Comment:
This doesn't feel like it belongs on `TagletWriterImpl`

Reply:
I'm okay with making it private, but this functionality is only currently needed in this class to replace code 50-ish lines above.